### PR TITLE
feat: index suggestion for the next wallet account name

### PR DESCRIPTION
**Pull request by saledjenic**
Whenever the user is about to add a new wallet account, we should call `GetIndexForNextAccountNameSuggestion` and offer an account name using the index from that call.

### CODE REVIEW

**Comment by J-Son89** [comment]
hmm, I won't review this until after our discussion with designers tomorrow 👍

File: multiaccounts/settings/database.go [hunk 1]
```diff
@@ -149,10 +149,11 @@ INSERT INTO settings (
   wallet_display_assets_below_balance,
   wallet_display_assets_below_balance_threshold,
   wallet_collectible_preferences_group_by_collection,
-  wallet_collectible_preferences_group_by_community
+  wallet_collectible_preferences_group_by_community,
+	last_used_wallet_account_name_index_suggestion
 ) VALUES (
 ?,?,?,?,?,?,?,?,?,?,?,?,?,?,
-?,?,?,?,?,?,?,?,?,'id',?,?,?,?,?,?,?,?,?,?,?,?)`,
+?,?,?,?,?,?,?,?,?,'id',?,?,?,?,?,?,?,?,?,?,?,?,?)`,
```
Comment (J-Son89) [inline multiaccounts/settings/database.go:156]: what is this code? 😅 not judging your code, just curious of what is going on here.

Comment (saledjenic) [reply multiaccounts/settings/database.go:156]: That's how we set initial settings, it's there for ages. 😃 
I suggested at some point in the past that we could refactor settings table to make it growing in height, not in width, but...

Comment (J-Son89) [reply multiaccounts/settings/database.go:156]: These are all positional args? Can they not be named args so we would at least know what is being set?

Comment (saledjenic) [reply multiaccounts/settings/database.go:156]: No, that's part of query. Look at the code.

Comment (J-Son89) [reply multiaccounts/settings/database.go:156]: Surely could have a helper function to build that query which does this though?

Comment (alaibe) [reply multiaccounts/settings/database.go:156]: if only we had invented the ORM for such simple query :trollface:

**Review by alaibe — APPROVED** [review]

**Comment by saledjenic** [comment]
@alaibe @J-Son89 @benjthayer @xAlisher should we go with this (having suggestions `Account 1`, `Account 2`, `Account 3`... or?

**Comment by benjthayer** [comment]
> @alaibe @J-Son89 @benjthayer @xAlisher should we go with this (having suggestions `Account 1`, `Account 2`, `Account 3`... or?

Given we are not entering a name into the field for the user and we are just showing prompt text, @xAlisher rightly suggested we could just go with `Account name` as the prompt text as opposed to locking ourselves into correctly indexing the account. This sounds fine from my perspective. It only really makes sense to index the name if it is a pre-populated suggestion which the user can actually choose to apply to their account.

cc @John-44

**Comment by saledjenic** [comment]
> It only really makes sense to index the name if it is a pre-populated suggestion which the user can actually choose to apply to their account.

@benjthayer your initial idea was to pre-populate the account name input field, when user opens add account popup, with the suggested account names, in the form of `Account 1, Account 2... Account 10...` with the aim that the user can add a new account quickly in one click without having to type the account name.

But I am also ok with the option to set `Account name` as placeholder text of the account name input field and force user to enter the name explicitly before adding an account.

Please confirm which one is our final decision?

**Comment by xAlisher** [comment]
> > @alaibe @J-Son89 @benjthayer @xAlisher should we go with this (having suggestions `Account 1`, `Account 2`, `Account 3`... or?
> 
> 
> 
> Given we are not entering a name into the field for the user and we are just showing prompt text, @xAlisher rightly suggested we could just go with `Account name` as the prompt text as opposed to locking ourselves into correctly indexing the account. This sounds fine from my perspective. It only really makes sense to index the name if it is a pre-populated suggestion which the user can actually choose to apply to their account.
> 
> 
> 
> cc @John-44 

Full support 🙌

**Comment by benjthayer** [comment]
> > It only really makes sense to index the name if it is a pre-populated suggestion which the user can actually choose to apply to their account.
> 
> @benjthayer your initial idea was to pre-populate the account name input field, when user opens add account popup, with the suggested account names, in the form of `Account 1, Account 2... Account 10...` with the aim that the user can add a new account quickly in one click without having to type the account name.
> 
> But I am also ok with the option to set `Account name` as placeholder text of the account name input field and force user to enter the name explicitly before adding an account.
> 
> Please confirm which one is our final decision?

Given what was discussed on the Wallet Brainstorm call where the indexing and auto-creation of the account name seemed to be causing issues, Alisher and I both agree that using prompt text instead makes sense and I think this aligns with the approach on mobile. Given we are using prompt text and the user has to insert a name anyway to proceed, there isn't much use in adding the index. So I'm +1 for just using the prompt text with no account index... so just prompt text `Account name` in the input field as opposed to an auto-generated name.

**Comment by saledjenic** [comment]
Closing this PR since the consensus is made that we will display just "Account name",  without indexing, as placeholder text for the account name input field.

**Comment by saledjenic** [comment]
@J-Son89 just to be consistent, please make sure that the mobile app works this way:
- when adding new Status account, the default wallet account, should be named "Account 1"
- opening the page for adding a new wallet account, the app should have an empty input field with the placeholder "Account name"
- no automatic indexing of newly added accounts based on the number of already added wallet accounts

**Comment by J-Son89** [comment]
> @J-Son89 just to be consistent, please make sure that the mobile app works this way:
> 
> - when adding new Status account, the default wallet account, should be named "Account 1"
> 
> - opening the page for adding a new wallet account, the app should have an empty input field with the placeholder "Account name"
> 
> - no automatic indexing of newly added accounts based on the number of already added wallet accounts 

Thanks Sale, will adjust!

**Pull request closed without merging**